### ISSUE TITLE
`autocomplete-preset-algolia` has missing peer dependency `algolia/client-search`

### ISSUE THREAD

**Issue by lex111** [algolia/docsearch#998]
When installing `docsearch/react` (eg, latest version), I noticed warning like this:

> @docusaurus/theme-search-algolia > @docsearch/react > @algolia/autocomplete-preset-algolia@1.0.0-alpha.35" has unmet peer dependency "@algolia/client-search@^4.5.1".

@francoischalifour could you please fix this?

**Comment by francoischalifour** [comment]
I believe this is fixed with newer versions of Autocomplete (underlying library of DocSearch): https://github.com/algolia/autocomplete.js/blob/971305a3250acf3f775207d172b6c68a37345028/packages/autocomplete-preset-algolia/package.json#L41

We haven't yet updated DocSearch to the latest Autocomplete version though, which should be why you see this warning.

I'll let you know here once we upgrade Autocomplete. Thanks for flagging.

**Comment by francoischalifour** [comment]
Actually to fix this we need to add `@algolia/client-search` to the [DocSearch peer dependencies](https://github.com/algolia/docsearch/blob/e7797a50d3dad6f74c484b8cf7cb28ee37412eab/packages/docsearch-react/package.json#L42-L46). We only import `@algolia/client-search` to use a type though.

I'm wondering if that's OK to ignore this npm warning?

**Comment by lex111** [comment]
Honesty, I don't know how best to do it, but I just wanted to get rid of such warnings, if possible.

**Comment by Haroenv** [comment]
The warning doesn't really have an impact on day-to-day development, since the dependency is already installed by another package (algoliasearch), so until we fix it (make the error no longer show up by updating autocomplete in docsearch), the warning will show but can be ignored!

**Comment by lex111** [comment]
Yes, I understand that this warning can be ignored, but I would like them not to exist at all.

By the way, I also noticed that you do not mark the latest versions with the appropriate label, please do not forget about this in the next releases.

![image](https://user-images.githubusercontent.com/4408379/107856242-ed64e080-6e37-11eb-92e0-50fd6e23e5ac.png)

**Comment by francoischalifour** [comment]
This is the npm behavior (or rather bug) when you start publishing versions with the alpha tag. The first version released will be tagged as `latest` on npm, so we cannot do anything about it. Once we release stable on the `latest` tag, it'll behave as expected.

**Comment by SimenB** [comment]
Running `yarn add  @docsearch/react` now (installing `3.5.2`) prints the following warnings:

```
warning "@docsearch/react > @algolia/autocomplete-preset-algolia@1.9.3" has unmet peer dependency "@algolia/client-search@>= 4.9.1 < 6".
warning "@docsearch/react > @algolia/autocomplete-core > @algolia/autocomplete-plugin-algolia-insights@1.9.3" has unmet peer dependency "search-insights@>= 1 < 3".
warning "@docsearch/react > @algolia/autocomplete-core > @algolia/autocomplete-shared@1.9.3" has unmet peer dependency "@algolia/client-search@>= 4.9.1 < 6".
```

Any plans to fix this?
